server/test/check.go

```diff
@@ -1810,6 +1807,87 @@ func TestCheckQueryAuthorizationModelsVersioning(t *testing.T, datastore storage
 	require.False(t, updatedResp.Allowed)
 }
 
+// TestCheckQueryIgnoresTuplesCurrentlyInvalid ensures that Check is ignoring tuples that were valid for some auth model but not
+// for the one passed in the request
+func TestCheckQueryIgnoresTuplesCurrentlyInvalid(t *testing.T, datastore storage.OpenFGADatastore) {
+	ctx := context.Background()
+	tracer := telemetry.NewNoopTracer()
+	meter := telemetry.NewNoopMeter()
+	logger := logger.NewNoopLogger()
+	store := ulid.Make().String()
+
+	// write one model with some tuples
+	oldModel := &openfgapb.AuthorizationModel{
```

Comment:
Can I suggest refactoring this test so it is more simplified and part of the main `TestCheckQuery` test suite?

Something like
```
func TestCheckQuery(...) {
    ...
    {
            name: "Invalid direct tuples are skipped over",
            typeDefinitions: []*openfgapb.TypeDefinition{
                    {
                            Type: "document",
                            Relations: map[string]*openfgapb.Userset{
                                    "viewer": typesystem.This(),
                            },
                    },
            },
            tuples: []*openfgapb.TupleKey{
                    tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
                    tuple.NewTupleKey("group:eng", "member", "jon"),
            },
            resolveNodeLimit: defaultResolveNodeLimit,
            request: &openfgapb.CheckRequest{
                    TupleKey: tuple.NewTupleKey("document:1", "viewer", "jon"),
            },
            response: &openfgapb.CheckResponse{
                    Allowed: false,
            },
    },
```

Reply:
I was wondering how these tests were going to pass when I noticed that the test is not calling `commands.WriteTuples` but `storage.Write` 🤣